Automattic/jetpack
# Social: Refactor SIG preview component

This change refactors the preview component a bit, so the values can come from props, instead of the hooks. It is also possible to turn off the debounce, as in the editor we won't need it anymore after introducing the modal in #31665.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create a new post with the Advanced plan and SIG turned on.
 * Changing the following should be reflected on the preview
  * Changing the title
  * Changing the description
  * Template
  * Image type, or image
  * Debouncing should work on text changes as before

* In `projects/js-packages/publicize-components/src/components/social-image-generator/panel/index.js` update the GeneratedImagePreview, and add properties to check that they overwrite the hook values. For example if `customText` is provided, it should be there all the time
